src/query/interpret/eval.hpp

```diff
@@ -476,24 +476,27 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
 
   TypedValue Visit(AllPropertiesLookup &all_properties_lookup) override {
     TypedValue::TMap result(ctx_->memory);
+    std::map<std::string, TypedValue> result_;
 
     auto expression_result = all_properties_lookup.expression_->Accept(*this);
     switch (expression_result.type()) {
       case TypedValue::Type::Null:
         return TypedValue(ctx_->memory);
       case TypedValue::Type::Vertex: {
-        const auto &vertex = expression_result.ValueVertex();
-        for (const auto &[property_id, value] : *vertex.Properties(view_)) {
+        const auto vertex = expression_result.ValueVertex();
+        auto properties = *vertex.Properties(view_);
```

Comment:
I think these two will create copies, right? Why not using reference here?

Reply:
Shouldn't RVO work here? But there is really no sense in using vertex as a local variable

Reply:
Right, I’d too expect RVO to prevent that. Anyway, I’ve accepted your suggestions and cut the local variables - they were only there for readability

Reply:
I think it should not kick in. As far as I understood, RVO kicks in when you have created local variable and then return that local variable. In this case, we return `vertex_v` from function which exists in `TypedValue`. And it that case I think it doesn't kick in.

Reply:
I think this is okay example: https://godbolt.org/z/ex4qc41v7